### ISSUE TITLE
Potential error in computing TV loss?

### ISSUE THREAD

**Issue by Wuziyi616** [apchenstu/TensoRF#25]
Hi, thanks for the great work! I'm reading the code and find something weird. In these [lines](https://github.com/apchenstu/TensoRF/blob/ba2c43e73c99cec12be1f4437cbef92ee3f794b9/train.py#L200-L209), you're computing the TV loss. At line203, `loss_tv` is first computed on density components, and then added to `total_loss` at line 204. However, at line207, you add the appearance component TV loss to the previous density TV loss:
```
loss_tv = loss_tv + tensorf.TV_loss_app(tvreg)*TV_weight_app
```
and add them together to the `total_loss` at line208:
```
total_loss = total_loss + loss_tv
```
If I understand correctly, you will punish density TV loss twice by doing so. While I understand we can adjust its loss weight to mitigate this issue, maybe you want to fix this.

Another problem is, you are doing decay of TV loss weight by `TV_weight_density *= lr_factor`. Did you find this better compared to e.g. using a consistent TV loss weight?

**Comment by Wuziyi616** [comment]
Besides the above issue, I found another problem in TVLoss computation, which I believe is a searious one @apchenstu 

In the TVLoss class you're dividing loss with the number of tensors, e.g. `count_w` [here](https://github.com/apchenstu/TensoRF/blob/ba2c43e73c99cec12be1f4437cbef92ee3f794b9/models/tensoRF.py#L158). However, as we can see from class `TensorVMSplit`, [line_coef](https://github.com/apchenstu/TensoRF/blob/ba2c43e73c99cec12be1f4437cbef92ee3f794b9/models/tensoRF.py#L158) has last_dim = 1. Therefore, the `count_w` will equal to 0, thus leading to a NaN in the [loss](https://github.com/apchenstu/TensoRF/blob/ba2c43e73c99cec12be1f4437cbef92ee3f794b9/utils.py#L151) after division. I actually encountered such NaN in my training... And I believe this can be easily fixed by taking a `min(count_w, 1)`.

**Comment by apchenstu** [comment]
Oh, they should be bugs, I will fix them and push a new version again, many thanks for pointing out than!

**Comment by Wuziyi616** [comment]
Thanks for the reply and thank you again for the great work!
